mitodl/mit-open
# Learning Resource cards, list view

### What are the relevant tickets?

Closes [Learning Resource Cards #4356](https://github.com/mitodl/hq/issues/4356)
Closes [Storybook stories for learning resource cards #384](https://github.com/mitodl/mit-open/issues/384)

### Description (What does it do?)

Implements the Learning Resource Card "list" variants for desktop and mobile.

https://www.figma.com/proto/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?page-id=107%3A2479&node-id=3375-49853&viewport=-15774%2C-1090%2C0.55&t=cX0nylncBbGUadPG-1&scaling=min-zoom&starting-point-node-id=2197%3A31345

Applies them to: 
- the Search page at `/search`
- the Learning Paths page at `/learningpaths`

Also updates the `Card` component in its link variant to reflect changes to the hover design and behavior, - https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=827-6168&m=dev:

- Whole card is clickable
- No shadow by default
- Shadow on hover
- No title underline on hover

### Screenshots (if appropriate):

Search Page:

<img width="861" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/a9f94da7-c1f3-4655-817a-38819e05961a">
<img width="856" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/1fe9fc18-6bbe-4fd4-a9b6-dcc6fd922748">

<img width="519" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/169d0548-c1df-413e-b327-8ba114e9874d">

------

Learning Paths page:

<img width="880" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/324d7fc1-021c-427f-8259-b1f9b2d4ff7b">

------

Card hover updates:

<img width="348" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/c18f61c7-4a77-4713-8d22-b7ae362a1368">
<img width="348" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/c433148e-53dd-47c1-bed8-429ee81780f2">



### How can this be tested?

Load the Search and Learning Path pages at `/search` and `/learningpaths` and ensure the list items display correctly on desktop and at mobile screen widths
- Click anywhere on card to navigate
- Default image is shown where none on learning resource
- Item count shown in learning path resources
- Certificate badge shown where resource is certificated
- Action buttons display correctly and open dialogs when clicked
   - Search page
     - Add to user list for authenticated user
     - Add to learning path for learning path editor
   - Learning Paths
      - Edit/delete menu 
- Skeleton load state displays correctly


### Additional Context




- Moves action buttons out of the parent Link container in the DOM structure and places them with css to avoid the HTML spec constraint that a button cannot be nested inside an anchor.
    - There's an issue that the action buttons can overlap the footer info. We can solve this with javascript, though it's a bit messy and could be better solved by design - min width on the container, another breakpoint between mobile and desktop, limits on the text displayed (display 3 letter month, display data only).
   
<img width="350" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/3b72f752-17e1-4421-903f-d0e0c8cbb7b2">


- The LearningResourceCard page-component (and LearningResourceCardTemplate) is still in use by the Learning Resource Details page. We'll need to refactor sortability across before we can remove it for this one.

- Adds a button variation for now that matches the design with `edge="none"`, `color="secondary"`. There are [CardActions button designs](https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=3152-18866&t=vAm3BtVv8nketftd-0) that we haven't yet modelled in code.

<img width="65" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/eee6667f-33a4-424c-b06e-b8a35ac38a2d">
